javascript/implementation/Database.ts

```diff
@@ -200,12 +202,50 @@ export class Database {
         return this.lastLinkToExtend;
     }
 
+    /**
+     * Reset all containers in the database to a previous time.
+     * @param toTime optional timestamp to reset to. If not provided, each container
+     * will be cleared.
+     * @param bundlerOrComment optional bundler to add this change to, or a string to
+     * add a comment to a new bundle.
+     */
+    async reset(
+        toTime?: AsOf,
+        bundlerOrComment?: Bundler | string
+    ): Promise<void> {
+        let immediate = false;
+        let bundler: Bundler;
+        if (bundlerOrComment instanceof Bundler) {
+            bundler = bundlerOrComment;
+        } else {
+            immediate = true;
+            bundler = new Bundler(bundlerOrComment);
+        }
+        const globalContainers: MuidTuple[] = [[-1, -1, Behavior.DIRECTORY]];
+        const containers = await this.store.getAllContainerTuples();
+
+        for (const muidTuple of containers) {
+            const container = await construct(this, muidTupleToMuid(muidTuple));
+            if (container instanceof Property) continue;
+            await container.reset({ toTime, bundlerOrComment: bundler });
+        }
+
+        for (const muidTuple of globalContainers) {
```

Comment:
It looks like `globalContainers` only consists of the global directory, but maybe should also include the other global containers.